src/Microsoft.DotNet.VersionTools/Automation/VersionsRepoUpdater.cs

```diff
@@ -96,18 +145,33 @@ public VersionsRepoUpdater(GitHubAuth gitHubAuth, GitHubProject project)
                     message += $" for {prereleaseVersion}";
                 }
 
-                await UpdateGitHubFileAsync(latestFilePath, prereleaseVersion, message);
+                GitTree tree = await client.PostTreeAsync(_project, masterSha, objects.ToArray());
+                GitCommit commit = await client.PostCommitAsync(_project, message, tree.Sha, new[] { masterSha });
+
+                // Only fast-forward. Don't overwrite other changes: throw exception instead.
+                await client.PatchReferenceAsync(_project, masterRef, commit.Sha, force: false);
```

Comment:
Isn't this going to potentially have race conditions? I suspect the files being written too are different for each official build so perhaps there shouldn't be any conflicts. At any rate we might need to consider some retry logic here if we start seeing failures.